vcr/vcr.go

```diff
@@ -262,7 +263,8 @@ func (c *vcr) Configure(config core.ServerConfig) error {
 		c.openidSessionStore = c.storageClient.GetSessionDatabase()
 	}
 	c.issuer = issuer.NewIssuer(c.issuerStore, c, networkPublisher, openidHandlerFn, didResolver, c.keyStore, c.jsonldManager, c.trustConfig)
-	c.verifier = verifier.NewVerifier(c.verifierStore, didResolver, c.keyResolver, c.jsonldManager, c.trustConfig)
+	c.verifierHttpClient = core.NewStrictHTTPClient(config.Strictmode, c.config.OpenID4VCI.Timeout, tlsConfig)
```

Comment:
I don't think strict HTTP client is required, since the credential is signed?

Side question: do we check that the status credential is signed by the credential (being checked)'s issuer?

Reply:
https is probably not required, same for the tlsConfig? and it also abuses the OpenID4VCI timeout. I wasn't sure so went with the safest option for now. CRLs are often served on http since they are signed and it solves a chicken-egg problem, but here that problem does not exist. Applying the node's strictmode setting made sense to me, but I am open to a broader solution to what is needed here.

Side question: no, will add to credentialStatus.Verify (the spec does not list this as requirement, but it is probably implied)

Reply:
Side question: no, ~~will add to credentialStatus.Verify (the spec does not list this as requirement, but it is probably implied)~~ the validate algorithm does not include a check that the `statusListCredential` and the `credentialToVerify` have the same issuer, only that they have the same `statusPurpose`.

Should we include this check? Without this check a vendor could have 1 statuslist for all the dids it manages (makes our lives a bit easier), but how should the trust in the issuer of the statuslist be derived? If we trust the credential issuer containing a `credentialStatus`, just trust the revocation mechanism it uses?

Reply:
IDeally you'd have multiple layers of protection, if issuer isn't checked and we have no HTTPS, you could spoof the endpoint (MITM) and you're in. If there's HTTPS, spoofing becomes a lot harder. If we have no HTTPS but the status credential issuer is checked, you need to steal a credential issuer private key and spoof DNS.

Enforcing HTTPS is probably the easiest fix here. Although I don't expect other DIDs than the credential issuer to sign the status credential? (maybe EBSI, but out of scope..)